community/modules/scripts/spack/templates/spack_install.tpl

```diff
@@ -0,0 +1,57 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+
+- name: Install necessary dependencies
+  hosts: localhost
+  tasks:
+  - name: Install git
+    ansible.builtin.package:
+      name:
+      - git
+      state: latest
+- name: Install Spack
+  hosts: localhost
+  vars:
+    install_dir: ${install_dir}
+    source_repo: ${source_repo}
+    spack_ref: ${spack_ref}
+    chmod_mode: ${chmod_mode}
+    chown_owner: ${chown_owner}
+    chgrp_group: ${chgrp_group}
+  tasks:
+  - name: Clones spack into installation directory
+    ansible.builtin.git:
```

Comment:
I'm curious what the behavior of this is upon a reboot when a user has made changes to the working copy / switched branches, etc, etc.

Reply:
Ditto when the user has changed group/owner/mode